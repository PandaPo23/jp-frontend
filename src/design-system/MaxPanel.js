import system from '../utils/System';
import FlexPanel from './FlexPanel';

const MaxPanel = system('MaxPanel', {
  extend: FlexPanel,
  width: 1,
  flex: 1,
  height: '100%',
  minHeight: 0,
});
MaxPanel.displayName = 'MaxPanel';

export default MaxPanel;
