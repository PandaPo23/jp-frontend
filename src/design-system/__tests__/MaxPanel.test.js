import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import MaxPanel from '../MaxPanel';

describe('MaxPanel component', () => {
  it('should render as MaxPanel', () => {
    const wrapper = mount(<MaxPanel />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
