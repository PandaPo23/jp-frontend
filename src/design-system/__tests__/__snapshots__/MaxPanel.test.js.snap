// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaxPanel component should render as MaxPanel 1`] = `
.emotion-0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  min-height: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 0;
}

<MaxPanel
  display="flex"
  flex={1}
  flexDirection="column"
  height="100%"
  minHeight={0}
  width={1}
>
  <div
    className="emotion-0"
  />
</MaxPanel>
`;
